### PR TITLE
add SafetyCommand class

### PR DESCRIPTION
**Pull request by ajnadel**
This PR:
- allows commands to cancel themselves if they are found to be unsafe

### CODE REVIEW

**Comment by ajnadel** [comment]
@c0d3rman the latest commit should satisfy your failure message request.

File: commands/SafetyCommand.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	protected final void initialize() {
+		if (isSafe()) {
+			initializeIfSafe();
+			return;
+		}
+		this.cancel();
+		if (reasonUnsafe == null) {
+			reasonUnsafe = "the required safety conditions haven't been met";
+		}
+		LogKitten.e("Command " + getName() + " cannot run because " + reasonUnsafe + ". Cancelling...");
```
Comment (c0d3rman) [inline commands/SafetyCommand.java]: Change `Command` to `SafetyCommand`.

Comment (ajnadel) [inline commands/SafetyCommand.java]: Fixed.

**Comment by c0d3rman** [comment]
:+1:

**Pull request merged**